src/treq/client.py

```diff
@@ -141,15 +146,18 @@ def delete(self, url, **kwargs):
         """
         return self.request('DELETE', url, **kwargs)
 
-    def request(self, method, url, **kwargs):
+    def request(self, method, url,
+                headers=None, params=None, data=None, files=None,
+                auth=None, cookies=None, allow_redirects=True,
```

Comment:
[`cookie_jar_from_dict()`](https://2.python-requests.org/en/master/api/#requests.cookies.cookiejar_from_dict) expects...  a `dict`. :]  The `files` change looks like it will retain previous behavior.

Reply:
Are you sure about that? I tested the call and it returns `==` equivalent CookieJars with None and empty dict.

Reply:
Yup, code is intentionally allowing for that, it's just not documented: https://2.python-requests.org/en/master/_modules/requests/cookies/#cookiejar_from_dict

Reply:
Welp, the source definitely has `if cookie_dict is not None:` in there so it'll work but it is not documented that `None` is an acceptable value.  Seems like minimal work and code to stick to the defined interface.  I checked against requests 3 and the code/docstrings are unchanged at this point compared to 2.

```python
if cookies is None:
    cookies = {}
```

or maybe

```python
if cookies is None:
    cookies = self._cookiejar
else:
    if not isinstance(cookies, CookieJar):
        cookies = cookiejar_from_dict(cookies)

    cookies = merge_cookies(self._cookiejar, cookies)
wrapped_agent = CookieAgent(self._agent, cookies)
```
---

Potential for another PR just to simplify this code a bit...

It seems like even using `cookiejar_from_dict()` is making things unnecessarily complicated.  `merge_cookies()` itself takes a `dict` or `CookieJar` as the second parameter so Treq doesn't need to be converting the dict to a `CookieJar` before merging.  Looks like `cookiejar_from_dict()` and `merge_cookies()` have been unchanged for about six years.

```python
if cookies is not None:
    merge_cookies(self._cookiejar, cookies)
wrapped_agent = CookieAgent(self._agent, self._cookiejar)
```

And since requests is `isinstance`ing all over, the `if cookies is not None:` here isn't even required...  just nice to stick to the documented interface I figure.  I suppose an argument could be made against doing this simplification.  It may use documented interfaces but if you don't like functions that take multiple types (`merge_cookies()` taking `dict` or `CookieJar`) you may prefer to not leverage that option even when it exists.

---

Sorry, didn't mean for this to grow quite this big...

Reply:
Oh, that does make sense :-). I added a small comment since it's not entirely evident that merge_cookies does modify the CookieJar, it could as well be functional and return a new merged CookieJar.
Good catch!

Reply:
Interestingly a test fails if used like this, I thought it was a no-brainer and had pushed without running trial locally, wonders of CI.
I adapted things to solve your concern about only relying on the documented interface (dicts) as opposed to rely on de-facto support in code for `None`.
This observation did cause some simplification which I think is still worth it.